### ISSUE TITLE
Type checking failures against numpy types

### ISSUE THREAD

**Issue by jenshnielsen** [microsoft/pyright#5879]
**Describe the bug**
The attached code typechecks with 1.1.321 but fails with 1.1.322 and later.
This seems to be because the output of np.max (b) is now detected as generic where before it used to be detected 
as Unknown. (for reference mypy 1.5.1 detects is as Any)
Is it intentionel that this is detected as generic?

**Code or Screenshots**

```python
import numpy as np
from typing_extensions import reveal_type

a: int = 1
array: np.ndarray = np.ndarray([1,2,3])
b = np.max(array)
reveal_type(b)
c = b > a

print(b)
```

numpy 1.25.2 
python 3.11 

**VS Code extension or command-line**
Commandline 1.1.322 and later

**Comment by erictraut** [comment]
Thanks for the bug report.

I've determined that the root cause for this bug was introduced a few versions earlier in 1.1.319 as part of the fix for [this issue](https://github.com/microsoft/pyright/issues/5404). The bug was masked until a later change in 1.1.322 exposed it.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.326, which I just published. It will also be included in a future release of pylance.

**Comment by jenshnielsen** [comment]
Thanks, I can confirm that this also fixed my more complex original issues.
